kvdb-rocksdb/src/lib.rs

```diff
@@ -152,47 +152,56 @@ impl CompactionProfile {
 pub struct DatabaseConfig {
 	/// Max number of open files.
 	pub max_open_files: i32,
-	/// Memory budget (in MiB) used for setting block cache size, write buffer size.
-	pub memory_budget: Option<usize>,
-	/// Compaction profile
+	/// Memory budget (in MiB) used for setting block cache size,
+	/// write buffer size for each column.
+	pub memory_budget: Vec<Option<usize>>,
+	/// Compaction profile.
 	pub compaction: CompactionProfile,
-	/// Set number of columns
-	pub columns: Option<u32>,
+	/// Specify the maximal number of info log files to be kept.
+	pub keep_log_file_num: i32,
 }
 
 impl DatabaseConfig {
 	/// Create new `DatabaseConfig` with default parameters and specified set of columns.
 	/// Note that cache sizes must be explicitly set.
 	pub fn with_columns(columns: Option<u32>) -> Self {
 		let mut config = Self::default();
-		config.columns = columns;
+		config.memory_budget.resize(columns.unwrap_or(0) as usize, None);
 		config
 	}
 
 	pub fn memory_budget(&self) -> usize {
-		self.memory_budget.unwrap_or(DB_DEFAULT_MEMORY_BUDGET_MB) * MB
+		if self.memory_budget.is_empty() {
+			return DB_DEFAULT_MEMORY_BUDGET_MB * MB;
+		}
+		self.memory_budget
+			.iter()
+			.map(|b| b.unwrap_or(DB_DEFAULT_MEMORY_BUDGET_MB) * MB)
+			.sum()
 	}
 
-	pub fn memory_budget_per_col(&self) -> usize {
-		self.memory_budget() / self.columns.unwrap_or(1) as usize
+	/// # Panics
+	///
+	/// If `col` is out of bounds.
+	pub fn memory_budget_per_col(&self, col: usize) -> usize {
+		self.memory_budget[col].unwrap_or(DB_DEFAULT_MEMORY_BUDGET_MB) * MB
```

Comment:
Let's do the multiplication with `MB` in the const definition.

Reply:
Not sure what you mean. Create a const function that multiples `usize` param by `MB`? What would be the benefit?

Reply:
No, I think I misread this, thought we did `DB_DEFAULT_MEMORY_BUDGET_MB * MB` in a few places and suggested we just do the multiplication when defining the const: `const DB_DEFAULT_MEMORY_BUDGET_MB: usize = 128 * MB;` but then we'd have to pass in the memory budget in bytes.
…so we need docs here to remind readers that the return value is in bytes. (And elsewhere that the `memory_budget` items are in mb).